packages/codemod/src/upgrade-legacy.ts

```diff
@@ -33,8 +33,12 @@ const isInternalBlitzMonorepoDevelopment = fs.existsSync(
 
 const upgradeLegacy = async () => {
   let isTypescript = fs.existsSync(path.resolve("tsconfig.json"))
-  let blitzConfigFile = `blitz.config.${isTypescript ? "ts" : "js"}`
-  let isLegacyBlitz = fs.existsSync(path.resolve(blitzConfigFile))
+  let existingBlitzConfigFiles = ["ts", "js"].map((e) => `blitz.config.${e}`).filter(fs.existsSync)
+  if (existingBlitzConfigFiles.length !== 1 || existingBlitzConfigFiles[0] === undefined) {
+    throw new ExpectedError("Could not identify Legacy Blitz Config file")
+  }
+  let isLegacyBlitz = true
```

Comment:
I think we should check `existingBlitzConfigFiles` to set `isLegacyBlitz` status, since we run a check here https://github.com/blitz-js/blitz/blob/4b8b7182aafb01558649b1ea69f83ef9e6029498/packages/codemod/src/upgrade-legacy.ts#L1190

Reply:
With the new detection, the `isLegacyBlitz` will always be true. This is due to the change, that I throw an error if I couldn't find the legacy configuration. In the previous version, this wasn't implemented.

I pushed a new commit with a change to make it a little bit cleaner with the reduce function, but I see now, that this would now also work if there is a `ts` and `js` file present and ignore the other one. Which may not be desired. Is there any behavior you have in mind, that would be the best fit for the config detection?

Reply:
> 

Regarding `isLegacyBlitz` will always be true. I get what you mean. What I was trying to avoid is duplicate logic as we're checking `isLegacyBlitz` later in the file and throwing an error.